### ISSUE TITLE
Design WebVR Precision Editor

### ISSUE THREAD

**Issue by alexmgrau** [SpliceVR/SpliceVREditor#4]
The precision editor will allow for editing within a node.

The WebVR interface should take advantage of the 360 space, adapt concepts from previous video editors (Premiere, iMovie, Blender, etc.), and concentrate on natural, streamlined interactions.

**Comment by grobm** [comment]
Hey Alex, I was wondering if you have looked at something like this?

https://github.com/casatt/html5-videoEditor

And consider integrating it? Also how are your planning to integrate transitions since they will need to take place outside the standard editor if you are planning to go multiple path, virtual tours schemas (old QTVR days)?

**Comment by alexmgrau** [comment]
Same as #3 but WebGL/WebVR is a definite for this one. I imagine for the precession editor that you can view the 360 video around you as you are editing. I am not sure if this is possible with this editor but it may serve as a good backbone.

As for transitions, there are a few possible types:
Node transitions: I believe you should be able to set in the precision editor or in the node editor. The UI is still up for discussion but I think if we use the right toolbar area to allow for the user to set the variables and actions to make the transition. 

Video transitions: If you mean a transition between videos (like cross fades, wipes, etc) I think these are a lower priority and will ideally be supported by the players themselves. I have been thinking about writing custom shaders that can mimic some of them in real time but the ideal starting place is to just have the content creators have all the transitions baked into the content. If they need to do a cross fade when going to the next branch of videos, then put the end of the current video at the beginning of each of the second videos and add the cross fade there. Not ideal but it is a starting place. 
tl;dr Video transitions are a hard question that needs more thought and research.
